LazyAlchemyLearner.lua

```diff
@@ -142,41 +187,164 @@ local function getItemLinkFromItemId(itemId)
 	return string.format("|H1:item:%d:%d:50:0:0:0:0:0:0:0:0:0:0:0:0:%d:%d:0:0:%d:0|h|h", itemId, 0, ITEMSTYLE_NONE, 0, 10000) 
 end
 
-local function getSolvent(proficiency, start)
+local function getSolvent(proficiency, startingPosition)
+	-- Check solvents
+	for i = startingPosition , proficiency + 1 do
+		-- poisons
+		local bag, bank, craft = GetItemLinkStacks(getItemLinkFromItemId(solvents[i]))
+		if((bag + bank + craft) > 0) then
+			return bag + bank + craft, solvents[i], i
+		end
 
-	for i = start, 1 , -1 do
-		local bag, bank, craft = GetItemLinkStacks(getItemLinkFromItemId(poison[i]))
-		return bag + bank + craft, poison[i],  i
+		-- potions
+		local bag, bank, craft = GetItemLinkStacks(getItemLinkFromItemId(solvents[i+9]))
```

Comment:
This construct is being used enough that I think it'll be useful to make a helper function for it
local function GetNumberOfAvailableItems(itemId)
  local bag, bank, craft = GetItemLinkStacks(getItemLinkFromItemId(itemId))
  return bag + bank + craft
end